src/Pester.Runtime.ps1

```diff
@@ -1486,6 +1486,7 @@ function Invoke-ScriptBlock {
                 finally {
                     if ($null -ne $______parameters.Teardown -and $______parameters.Teardown.Length -gt 0) {
                         if ($______parameters.EnableWriteDebug) { &$______parameters.WriteDebug "Running inner teardowns" }
+                        if ($______parameters.MoveBetweenScopes) { & $______parameters.SwitchTimerUserCode }
```

Comment:
Might not be needed, but added for consistency. Thoughts?

Looks like the issue is caused by `Invoke-TestItem` calling `Invoke-ScriptBlock` with `-MoveBetweenScopes`. When the test + AfterEach is finished, the clocks are switched to FrameworkDuration before processing outerTeardown (AfterAll) for the block.